### PR TITLE
feat(addons): Respect `.gitignore` in eslint addon

### LINKED ISSUE #334: feat: Add `.vercel` to eslint addon ignores
The `@sveltejs/adapter-vercel` defaults it's output to `.vercel` in the root of the project. The current ignore rules don't ignore this.

Current:
```ts
const ignoresConfig = object.create({
	ignores: common.expressionFromString('["build/", ".svelte-kit/", "dist/"]')
});
```

Happy to fill out a PR if this makes sense to add!

### PR DESCRIPTION
**Pull request by ieedan**
Fixes #334 

Searched the docs and tried the adapters out ended up also adding `.wrangler` and `.netlify`

### CODE REVIEW

**Comment by AdrianGonz97** [comment]
thank you! however I'm not sure if having to manually update the `ignores` property for every new entry is the best solution (also, TIL that eslint v9 no longer takes the `.gitignore` file into consideration, which is fun). 

perhaps it would be nicer to have the `.gitignore` automatically included using the [`@eslint/compat` package](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files)? 

It can be implemented as so:
```js
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));
export default {
 // ...
 includeIgnoreFile(gitignorePath)
}
```

thoughts? @manuel3108 @benmccann

**Comment by ieedan** [comment]
> thank you! however I'm not sure if having to manually update the `ignores` property for every new entry is the best solution (also, TIL that eslint v9 no longer takes the `.gitignore` file into consideration, which is fun).
> 
> perhaps it would be nicer to have the `.gitignore` automatically included using the [`@eslint/compat` package](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files)?
> 
> It can be implemented as so:
> 
> ```js
> import { fileURLToPath } from "node:url";
> import { includeIgnoreFile } from "@eslint/compat";
> 
> const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));
> export default {
>  // ...
>  includeIgnoreFile(gitignorePath)
> }
> ```
> 
> thoughts? @manuel3108 @benmccann

I don't think this is part of the .gitignore? It should be the `ignores` key:
```jsonc
{
	ignores: ['build/', '.svelte-kit/', 'dist/', '.vercel/']
}
```

**Comment by manuel3108** [comment]
Yeah, i think that approach would make sense, although we will have basically the same problem in the `.gitignore` file then. But personally, I don't think this should be implemented in a "hardcoded" way. I would prefer to see this implemented together with #248

**Comment by manuel3108** [comment]
> the `.gitignore` already includes all of the different adapter output dirs anyway (or at least it _should_, in the case of netlify which needs to be fixed)

At this point for me it doesn't look like it's including the vercel output. So that's also missing. But I wouldn't be in favor of adding that globally to the templates, as explained above.

**Comment by AdrianGonz97** [comment]
> At this point for me it doesn't look like it's including the vercel output. So that's also missing.

here's the currently generated `.gitignore`:
```
node_modules

# Output
.output
.vercel
/.svelte-kit
/build

# OS
.DS_Store
Thumbs.db

# Env
.env
.env.*
!.env.example
!.env.test

# Vite
vite.config.js.timestamp-*
vite.config.ts.timestamp-*

```

the only one that's missing is `.netlify` (and maybe wrangler? i normally use `@sveltejs/adapter-cloudflare` which outputs into `.svelte-kit/cloudflare` instead, so idk). 

> But I wouldn't be in favor of adding that globally to the templates, as explained above.

yea but that would only be helpful for people who are changing adapters via `add` and not manually. why not just get them all in one go instead? we're already most of the way there anyway since we're only missing 1 or 2 entries.

**Comment by ieedan** [comment]
I just went through and ran all of the adapters to see what they output since I don't think any of those outputs are actually in the docs for the adapters at least that I could find.

**Comment by AdrianGonz97** [comment]
here are the generated dirs in the root:
```shell
@sveltejs/adapter-node - /build
@sveltejs/adapter-static - /build
@sveltejs/adapter-cloudflare - /.wrangler 
@sveltejs/adapter-cloudflare-workers /.wrangler
@sveltejs/adapter-netlify - /.netlify
@sveltejs/adapter-vercel - /.vercel
```

not sure where `.output` dir that already exists in the `.gitignore` is supposed to be for, but we only need to add `.wrangler` and `.netlify` to the gitignore.

**Comment by ieedan** [comment]
Is there a reason there can't just be a exported constant somewhere that has all of these ignores to be used everywhere?

File: .changeset/beige-peas-swim.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/addons': patch
```
Comment (benmccann) [inline .changeset/beige-peas-swim.md:2]:
```suggestion
'@sveltejs/addons': minor
```

File: .changeset/beige-peas-swim.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Add `@sveltejs/<adapter>` out directories to eslint ignores.
```
Comment (benmccann) [inline .changeset/beige-peas-swim.md:5]:
```suggestion
feat: add `@sveltejs/<adapter>` out directories to eslint ignores
```

File: .changeset/beige-peas-swim.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/addons': minor
```
Comment (AdrianGonz97) [inline .changeset/beige-peas-swim.md:2]: sorry! `addons` is a private package so we'll need to increment `sv` instead, which is also in pre-release so it should also be a `patch`
```suggestion
'sv': patch
```

**Comment by AdrianGonz97** [comment]
> Is there a reason there can't just be a exported constant somewhere that has all of these ignores to be used everywhere?

not exactly sure I follow - who should be exporting these? the individual adapter packages?

**Comment by ieedan** [comment]
> > Is there a reason there can't just be a exported constant somewhere that has all of these ignores to be used everywhere?
> 
> not exactly sure I follow - who should be exporting these? the individual adapter packages?

I don't think it being a separate package would be practical but it may be nice to have an internal package to use for it? Like a @sveltejs/constants or something idk.

**Comment by AdrianGonz97** [comment]
eh, that would be pretty niche and not particularly helpful when it comes to the `gitignore` file since there's no way to 'import' those constants into it

anyway, I think we're in agreement that implementing https://github.com/sveltejs/cli/pull/335#issuecomment-2504356903 would be good, so feel free to add it to this PR!

**Comment by ieedan** [comment]
I meant more it could just be passed around to build whatever files need it.

Either way let me wrap a few things up and I can give it a shot!

**Comment by ieedan** [comment]
This should be it but I cannot seem to get it to build or run tests on my machine I am probably just doing something wrong.

**Review by manuel3108 — APPROVED** [review]
Thanks, so much! This seems to be working great!

The only thing im unsure about, is the placement of the `includeIgnoreFile` call. According to the [readme](https://www.npmjs.com/package/@eslint/compat) I would understand that this should be placed first in the eslint array, but currently it's placed last. In my local testing though, this didn't make any difference at all.

**Comment by ieedan** [comment]
> Thanks, so much! This seems to be working great!
> 
> The only thing im unsure about, is the placement of the `includeIgnoreFile` call. According to the [readme](https://www.npmjs.com/package/@eslint/compat) I would understand that this should be placed first in the eslint array, but currently it's placed last. In my local testing though, this didn't make any difference at all.

Yeah if you would rather have it at the beginning I can move it no problem at all!

**Comment by ieedan** [comment]
Also should I update the template git ignores with the output files mentioned above?

**Comment by AdrianGonz97** [comment]
Not at the moment, but we have plans on addressing issues similar to this in the future

**Pull request merged**